cmd/root.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/spec-tacles/spectacles.go/broker"
+	"github.com/spec-tacles/spectacles.go/gateway"
+	"github.com/spec-tacles/spectacles.go/rest"
+	"github.com/spec-tacles/spectacles.go/types"
+	"github.com/spf13/cobra"
+	"log"
+	"os"
+	"time"
+)
+
+var amqpUrl string
+var amqpGroup string
+var token string
+var shardCount int
+
+var logger = log.New(os.Stdout, "[CMD] ", log.Ldate|log.Ltime|log.Lshortfile)
+
+func main() {
+	rootCmd.Execute()
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "spectacles",
+	Short: "Connects to the Discord websocket API using spectacles.go",
+	Run: func(cmd *cobra.Command, args []string) {
+		amqp := broker.NewAMQP(amqpGroup, "", func(string, []byte) {})
+		done := make(chan bool)
+		go tryConnect(amqp, done)
+
+		manager := gateway.NewManager(&gateway.ManagerOptions{
+			ShardOptions: &gateway.ShardOptions{
+				Identify: &types.Identify{
+					Token: token,
+				},
+			},
+			OnPacket: func(shard int, d *types.ReceivePacket) {
+				pk, err := json.Marshal(&struct {
+					Shard int         `json:"shard_id"`
+					Data  interface{} `json:"data"`
+				}{shard, d.Data})
+				if err != nil {
+					return
+				}
+
+				select {
+				case done <- true:
```

Comment:
I'm not sure how this works, but I think this would deadlock the application since there's nothing reading from the `done` channel.

https://play.golang.org/p/qph_WCDOofJ vs https://play.golang.org/p/9hewoYPlDjZ